code/Gerrit.md

```diff
@@ -0,0 +1,282 @@
+
+```
+git rebase --continue
+```
+
+Now, your commit history looks the same as before, and you can push to Gerrit, and the correct reviews will all be updated.
+
+## Setting up Gerrit for the first time
+
+In this example, we are setting up Gerrit the main repository, https://github.com/storj/storj/. The steps should be almost identical for other repositories.
+
+### Permissions
+
+Before requesting Gerrit access in Slack, you will need to log in at https://review.dev.storj.io/ with your Github account. Once you have done this, share your Github email and username, and ask for review and submit permissions in the #gerrit Slack channel. When you have permissions, you can verify that you have submit access by going to any open changeset, and checking for a "CODE-REVIEW+2" button in the top right. If you don't see this, you still need to be added to the Reviewers group. 
+
+Next, set up your SSH key in Gerrit by going to your settings: https://review.dev.storj.io/settings/#SSHKeys. If you have already set up SSH with Github, this should be easy. Otherwise, see Github's docs for
+
+* [checking for existing ssh keys](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/checking-for-existing-ssh-keys)
+* [generating a new ssh key (if no existing ones)](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)
+* [adding an ssh key to your account](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account) - the only difference here is that you are adding the ssh key in your Gerrit settings rather than your Github settings.
+
+### Setting up an existing storj/storj clone to work with Gerrit
+
+#### Automatic setup
+
+Run
+
+```
+curl -L [storj.io/clone](http://storj.io/clone) | sh
+```
+
+from inside your local storj/storj repository.
+
+#### Manual setup
+
+Run the following command from your local storj/storj repository in order to add a hook which executes whenever a commit message is written:
+
+```
+scp -p -P 29418 <yourusername>@review.dev.storj.io:hooks/commit-msg ".git/hooks/"
+```
+
+This will add a line that looks like `Change-Id: I989b8a46c0e97c1278517e1de07fe42d5950cf54` to the end of each commit message you create, allowing Gerrit to track a single changeset, even as the commit hash, message, and files change.
+
+Now, add the `gerrit` remote path, allowing you to push to gerrit:
+
+```
+git remote add gerrit ssh://<yourusername>@review.dev.storj.io:29418/storj/storj
+```
+
+#### Next steps
+
+You should be all set now! When you have a commit to push for review, use one of the following commands (remember to `git pull --rebase origin main`, or the change may not be pushed successfully:
+
+* `git push gerrit HEAD:refs/for/main`
+* `git push gerrit HEAD:refs/for/main%wip` - if you do this, the changeset will be created in gerrit, but it will not trigger a build until you take it out of "wip"
```

Comment:
Let' s mention the "ready" push https://gerritcodereview-test.gsrc.io/marking-a-change-as-a-wip.html, but also the "private" https://gerritcodereview-test.gsrc.io/marking-a-change-as-private.html